cmake/Scripts/format.cmake

```diff
@@ -0,0 +1,34 @@
+# Get executable
+if(NOT CLANG_FORMAT_EXECUTABLE)
+    set(CLANG_FORMAT_EXECUTABLE clang-format)
+endif()
+
+# Check executable exists
+if(NOT EXISTS ${CLANG_FORMAT_EXECUTABLE})
+    find_program(CLANG_FORMAT_EXEC_TEMP ${CLANG_FORMAT_EXECUTABLE})
+    if(CLANG_FORMAT_EXEC_TEMP)
+        set(CLANG_FORMAT_EXECUTABLE ${CLANG_FORMAT_EXEC_TEMP})
+        unset(CLANG_FORMAT_EXEC_TEMP)
+    else()
+        message(FATAL_ERROR "Unable to find clang-format executable: \"${CLANG_FORMAT_EXECUTABLE}\"")
+    endif()
+endif()
+
+# Check executable version
+execute_process(COMMAND ${CLANG_FORMAT_EXECUTABLE} --version OUTPUT_VARIABLE CLANG_FORMAT_VERSION)
+string(REGEX MATCH "clang-format version ([0-9]+)" CLANG_FORMAT_VERSION ${CLANG_FORMAT_VERSION})
+unset(CLANG_FORMAT_VERSION)
+if(CMAKE_MATCH_1 EQUAL 12 OR CMAKE_MATCH_1 EQUAL 13)
+    message(STATUS "Using clang-format version ${CMAKE_MATCH_1}")
+else()
+    message(FATAL_ERROR "clang-format version ${CMAKE_MATCH_1} is not supported")
+endif()
```

Comment:
Do we want a closed set of version numbers or are we better off future proofing ourselves by requiring a minimum but accepting newer versions? clang-format-15 is already in development so it's safe to say that we're not far off from somebody attempting to use a newer version. I'm of the opinion that we should try to support to those newer versions until we run into issues.

Reply:
That would be fine by me, I was more concerned about some non-trivial change that could break it without overriding some option.

Reply:
Are these releases generally backwards-compatible? If yes, then I see no reason to prevent newer versions.

Reply:
> Are these releases generally backwards-compatible?

While older versions weren't due to non-toggleable bug fixes, 12 and 13 were. I don't suspect this will change in 14/15, but if it does then I'm sure we can catch it and change it. I'll go ahead and change it to `>12`.